File: RecruitmentATS/src/RecruitmentATS.Domain/Entities/JobPosting.cs
using RecruitmentATS.Domain.Common;
using RecruitmentATS.Domain.Enums;

namespace RecruitmentATS.Domain.Entities;

public class JobPosting : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Requirements { get; set; } = string.Empty;
    public string Location { get; set; } = string.Empty;
    public string? SalaryRange { get; set; }
    public string? Department { get; set; }
    public string? ContractType { get; set; }
    public JobPostingStatus Status { get; set; } = JobPostingStatus.Draft;
    public DateTime? ExpiresAt { get; set; }

    public ICollection<Candidate> Candidates { get; set; } = new List<Candidate>();
    public ICollection<JobPostingTag> JobPostingTags { get; set; } = new List<JobPostingTag>();
}

File: recruitment-frontend/src/types/jobPosting.ts
export interface JobPosting {
  id: string;
  title: string;
  description: string;
  requirements: string;
  location: string;
  salaryRange: string | null;
  department: string | null;
  contractType: string | null;
  status: string;
  createdAt: string;
  expiresAt: string | null;
  candidateCount: number;
  tags: string[];
}

export interface CreateJobPostingRequest {
  title: string;
  description: string;
  requirements: string;
  location: string;
  salaryRange?: string;
  department?: string;
  contractType?: string;
  expiresAt?: string;
  tags: string[];
}

export interface UpdateJobPostingRequest {
  title?: string;
  description?: string;
  requirements?: string;
  location?: string;
  salaryRange?: string;
  department?: string;
  contractType?: string;
  status?: string;
  expiresAt?: string;
  tags?: string[];
}